### ISSUE TITLE
Intl::getRegionBundle()->getCountryNames() gets in infinite loop

### ISSUE THREAD

**Issue by eugenmihailescu** [symfony/symfony#32028]
**Symfony version(s) affected**: 3.4.4

**Description**  
Calling Intl::getRegionBundle()->getCountryNames() ends up - for one reason or another - in an infinite loop.

**How to reproduce**  

By calling the `Intl::getRegionBundle()->getCountryNames()` this will call `RegionDataProvider::getNames` which in turn calls the `BundleEntryReader::readEntry`.

In that readEntry `while` loop the `$currentLocale` oscillate continuously between these three values:

- "en"
- "root"
- `false`

then again and again and again, the loop never ends and thus sooner or later the system memory will be exceeded: 
```php
 // Remember which locales we tried
$testedLocales[] = $currentLocale;
```

**Possible Solution**  
To prevent such situation I have double-checked if the `currentLocale` was seen before, in which case it is a sign of `something rotten`, thus I exit the loop forcebly:
```php
if(in_array($currentLocale, $testedLocales)){
      break;
}
 // Remember which locales we tried
$testedLocales[] = $currentLocale;

```

**Additional context**  
PHP 7.2.19
Linux 5.1.6 on Gentoo
dev-libs/icu 64.2

It run well for two years until I updated the OS packages, so this symptom might appear rather in odd situation. However some form of infinite loop prevention would be helpful.

**Comment by ro0NL** [comment]
Wasnt there some issue due a change in the php intl extension... ie https://github.com/symfony/symfony/issues/31087

**Comment by dmaicher** [comment]
@eugenmihailescu can you confirm that the issue is fixed with the latest version `v3.4.28`?

**Comment by eugenmihailescu** [comment]
> @eugenmihailescu can you confirm that the issue is fixed with the latest version `v3.4.28`?

Yes, the issue is fixed in v3.4.28 :)
